vividus-plugin-rest-api/src/test/java/org/vividus/http/HttpRequestExecutorTests.java

```diff
@@ -0,0 +1,184 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.http;
+
+import static com.github.valfirst.slf4jtest.LoggingEvent.info;
+import static java.util.Collections.singletonList;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.ArgumentMatchers.nullable;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+
+import com.github.valfirst.slf4jtest.TestLogger;
+import com.github.valfirst.slf4jtest.TestLoggerFactory;
+import com.github.valfirst.slf4jtest.TestLoggerFactoryExtension;
+
+import org.apache.http.ConnectionClosedException;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.CookieStore;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.protocol.HttpContext;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.vividus.http.client.HttpResponse;
+import org.vividus.http.client.IHttpClient;
+import org.vividus.http.exception.HttpRequestBuildException;
+import org.vividus.softassert.ISoftAssert;
+
+@ExtendWith({ MockitoExtension.class, TestLoggerFactoryExtension.class })
+class HttpRequestExecutorTests
+{
+    private static final String URL = "http://www.example.com/";
+    private static final String LOG_MESSAGE_FORMAT = "Response time: {} ms";
+    private static final long RESPONSE_TIME_IN_MS = 100;
+
+    @Mock
+    private IHttpClient httpClient;
+
+    @Mock
+    private HttpTestContext httpTestContext;
+
+    @Mock
+    private ISoftAssert softAssert;
+
+    private final TestLogger logger = TestLoggerFactory.getTestLogger(HttpRequestExecutor.class);
+
+    @InjectMocks
+    private HttpRequestExecutor httpRequestExecutor;
+
+    @Test
+    void testExecuteHttpRequestFirstTime() throws IOException
+    {
+        mockHttpResponse(HttpRequestBase.class, URL);
+        httpRequestExecutor.executeHttpRequest(HttpMethod.GET, URL, Optional.empty());
+        verify(httpTestContext).getRequestHeaders();
+        verify(httpTestContext).getRequestEntity();
+        assertThat(logger.getLoggingEvents(),
+                equalTo(singletonList(info(LOG_MESSAGE_FORMAT, RESPONSE_TIME_IN_MS))));
+        verify(httpTestContext).releaseRequestData();
```

Comment:
I propose to add verifyNoMoreInteractions here and in other places

Reply:
Where else it could be interacted with?

Reply:
I meant all the mocks used in tests, verifyNoMoreInteractions takes varargs as parameter

Reply:
Won't it be overhead for all tests? In such logic we are to update all unit-tests that we have?

Reply:
Current verification already checks exact mock method is used only one time

Reply:
verifyNoMoreInteractions checks that there are no more interactions with passed as argument mocks were made, except those that we verified